compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -446,10 +448,77 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                 }
             }
         }
+
         err.span_label(decl_span, "binding declared here but left uninitialized");
+        if show_assign_sugg {
+            struct LetVisitor {
+                decl_span: Span,
+                ty_span: Option<Span>,
+            }
+
+            impl<'v> Visitor<'v> for LetVisitor {
+                fn visit_stmt(&mut self, ex: &'v hir::Stmt<'v>) {
+                    if self.ty_span.is_some() {
+                        return;
+                    }
+                    if let hir::StmtKind::Local(hir::Local {
+                            span, init: None, ty: Some(ty), ..
+                        }) = &ex.kind && span.contains(self.decl_span) {
+                            self.ty_span = Some(ty.span);
+                    }
+                    hir::intravisit::walk_stmt(self, ex);
+                }
+            }
+
+            let mut visitor = LetVisitor { decl_span, ty_span: None };
+            visitor.visit_body(&body);
+            if let Some(ty_span) = visitor.ty_span {
+                self.suggest_assign_value(&mut err, moved_place, ty_span);
+            }
+        }
         err
     }
 
+    fn suggest_assign_value(
+        &self,
+        err: &mut Diagnostic,
+        moved_place: PlaceRef<'tcx>,
+        ty_span: Span,
+    ) {
+        let ty = moved_place.ty(self.body, self.infcx.tcx).ty;
+        debug!("ty: {:?}, kind: {:?}", ty, ty.kind());
+
+        let assign_value = match ty.kind() {
+            ty::Int(_) | ty::Uint(_) => format!("0"),
+            ty::Float(_) => format!("0.0"),
+            ty::Bool => format!("false"),
+            ty::Never | ty::Error(_) => "".to_string(),
+            ty::Adt(def, _substs) => {
+                if Some(def.did()) == self.infcx.tcx.get_diagnostic_item(sym::Vec) {
+                    format!("vec![]")
+                } else if let Some(default_trait) = self.infcx.tcx.get_diagnostic_item(sym::Default) &&
+                    self.infcx.tcx.infer_ctxt().enter(|infcx| {
+                        infcx.type_implements_trait(default_trait, ty, ty::List::empty(), self.param_env).may_apply()
+                    }) {
+                    format!("Default::default()")
+                } else {
+                    format!("todo!()")
+                }
+            }
+            _ => format!("todo!()"),
+        };
+
+        if assign_value.is_empty() {
+            return;
+        }
+        err.span_suggestion_verbose(
+            ty_span.shrink_to_hi(),
+            format!("consider assigning a default value"),
```

Comment:
Sorry to come back to this, but now when we suggest a value that’s not a `Default::default`, we should probably say just

> consider assigning a value

I'd imagine we could remove `default` for the `Default::default()` case too.